### DISCUSSION TITLE
Stepper - How to remember state of step without initialization each time

### DISCUSSION THREAD

**Discussion by vrjaroslav** [primefaces/community#1830] [PrimeNG] [answered]
Hello,

Is there any way for a stepper to remember the state of each step?

For example, I have a component named 'FirstComponent' in step 1, which is correctly initialized the first time. However, when I click on the next step (step 2) and then return back to the first step, the component initializes again. But I want to maintain the state that was there before I moved to step 2. So, the first component should not initialize again.

Is this possible? Can somebody please help me?

Here is example: [https://stackblitz.com/edit/fhpbyg?file=src%2Fapp%2Fdemo%2Fstepper-basic-demo.html,src%2Fapp%2Ffirst-component%2Ffirst-component.component.html,src%2Fapp%2Ffirst-component%2Ffirst-component.component.ts,src%2Fapp%2Fapp.module.ts](https://stackblitz.com/edit/fhpbyg?file=src%2Fapp%2Fdemo%2Fstepper-basic-demo.html,src%2Fapp%2Ffirst-component%2Ffirst-component.component.html,src%2Fapp%2Ffirst-component%2Ffirst-component.component.ts,src%2Fapp%2Fapp.module.ts)

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Thanks for the valuable opinion, the component wasn't designed in the way you mentioned. However, a workaround **may** be possible by utilizing viewChild.

**Reply by vrjaroslav** [reply]
Hi @cetincakiroglu, thanks for your answer and suggested workaround
